server/src/analyser.ts

```diff
@@ -399,6 +399,67 @@ export default class Analyzer {
     return name
   }
 
+  /**
+   * Find a block of comments above a line position
+   */
+  public commentsAbove(uri: string, line: number): string | null {
+    const doc = this.uriToTextDocument[uri]
+
+    const commentBlock = []
+
+    // start from the line above
+    let commentBlockIndex = line - 1
+
+    // check if that line starts with a comment
+    // its possible that some lines
+    // might have whitespace before the comment
+    // begins, so check by iterating from
+    // the start of the line
+    const startsWithAComment = (line: string): boolean => {
```

Comment:
It would be more natural to use a regular expression here.

```
const commentRegExp = /^\s*\#\s*(.*)/g

function getComment(line: string): null | string {
  const matches = commentRegExp.exec(' # this is a comment . ')
  return matches ? matches[1].trim() : null
}
```

Then you can run the while loop until it returns `null`. This also does the newline removal and trimming.